lib/Semantics/expression.cpp

```diff
@@ -238,8 +238,10 @@ MaybeExpr ExpressionAnalyzer::CompleteSubscripts(ArrayRef &&ref) {
   if (subscripts == 0) {
     // nothing to check
   } else if (subscripts != symbolRank) {
-    Say("Reference to rank-%d object '%s' has %d subscripts"_err_en_US,
-        symbolRank, symbol.name(), subscripts);
+    if (symbolRank != 0) {
+      Say("Reference to rank-%d object '%s' has %d subscripts"_err_en_US,
+          symbolRank, symbol.name(), subscripts);
```

Comment:
I am a bit lost with the semantic check status for assumed-rank array, I have them in mind because I just reviewed #1010 .  The code above would still emit something weird for them:
```
function foo(b)
  integer :: foo
  integer :: b(..)
  foo = b(2, 2)
end function
```
`error: Reference to rank-1 object 'b' has 2 subscripts` is emitted because `symbol.Rank()` for assumed-rank returns 1.

If we are not yet enforcing `C838` then that may just be a TODO and please ignore my comment.